data_processing/wdl/scatterGuppy.wdl

```diff
@@ -0,0 +1,348 @@
+				guppy_version = guppy_version
+		}
+
+		call concatenateSummary {
+			input:
+				files = guppyGPU.summary,
+				sample_name = sample_name,
+				guppy_version = guppy_version
+		}
+
+	}
+
+	output {
+		Array[File] bams = concatenateBam.concatenatedBam
+		Array[File] fastqs = concatenateFastq.concatenatedFastq
+		Array[File] summaries = concatenateSummary.concatenatedSummary
+	}
+	
+}
+
+task concatenateBam {
+	input {
+		Array[File] files
+		
+		String sample_name
+		String guppy_version
+
+		String dockerImage = "tpesout/megalodon:latest"
+
+		Int preempts = 3
+		Int memSizeGB = 8
+		Int threadCount = 3
+		Int diskSizeGB = 500
+	}
+	
+
+	command {
+		samtools merge -o "${sample_name}_${guppy_version}.bam" ${sep=" " files}
+	}
+
+	output {
+		File concatenatedBam = "${sample_name}_${guppy_version}.bam"
+	}
+
+	runtime {
+		memory: memSizeGB + " GB"
+		cpu: threadCount
+		disks: "local-disk " + diskSizeGB + " SSD"
+		docker: dockerImage
+		preemptible : preempts
+	}
+}
+
+
+task concatenateFastq {
+	input {
+		Array[File] files
+		String sample_name
+		String guppy_version
+
+
+		String dockerImage = "tpesout/megalodon:latest"
+
+		# runtime
+		Int preempts = 3
+		Int memSizeGB = 8
+		Int threadCount = 3
+		Int diskSizeGB = 500
+	}
+	
+
+	command {
+		cat ${sep=" " files} | gzip -c > "${sample_name}_${guppy_version}.fastq.gz"
+	}
+
+	output {
+		File concatenatedFastq = "${sample_name}_${guppy_version}.fastq.gz"
+	}
+
+	runtime {
+		memory: memSizeGB + " GB"
+		cpu: threadCount
+		disks: "local-disk " + diskSizeGB + " SSD"
+		docker: dockerImage
+		preemptible : preempts
+	}
+}
+
+task concatenateSummary {
+	input {
+		Array[File] files
+		String sample_name
+		String guppy_version
+
+
+		String dockerImage = "tpesout/megalodon:latest"
+
+		# runtime
+		Int preempts = 3
+		Int memSizeGB = 8
+		Int threadCount = 3
+		Int diskSizeGB = 500
+	}
+	
+
+	command {
+		cat ${sep=" " files} > "tmp.txt"
+		# remove duplicate headers
+		awk 'NR==1 || !/^filename/' "tmp.txt" > "${sample_name}_${guppy_version}_sequencing_summary.txt"
+	}
+
+	output {
+		File concatenatedSummary = "${sample_name}_${guppy_version}_sequencing_summary.txt"
+	}
+
+	runtime {
+		memory: memSizeGB + " GB"
+		cpu: threadCount
+		disks: "local-disk " + diskSizeGB + " SSD"
+		docker: dockerImage
+		preemptible : preempts
+	}
+}
+
+
+task splitFast5s {
+	input {
+		File file_to_split_tar
+		Int desired_size_GB
+
+		String dockerImage = "jiminpark/guppy-wdl:latest" 
+
+		Int preempts = 3
+		Int memSizeGB = 8
+		Int extraDisk = 5
+		Int threadCount = 2
+	}
+
+	Int file_size = ceil(size(file_to_split_tar, "GB"))
+	Int diskSizeGB = 3 * file_size + extraDisk
+
+
+	command <<<
+		## Extract tar file to 
+		mkdir tmp
+		
+		# place all extracted files into directory tmp
+		tar xvf "~{file_to_split_tar}" --directory tmp
+		rm ~{file_to_split_tar}
+
+		mkdir input
+
+		# filter out only fast5 files
+		for FILE in `find tmp/ -name "*.fast5"`
+		do
+			mv $FILE input
+		done
```

Comment:
Do you need this code block? You are just creating a second copy of the fast5 files. Instead, you could just make the line below
```
for FILE in `ls input`
``` 
into
```
for FILE in `find tmp/ -name "*.fast5"`
```
If you do that, you don't need to make the second copy of the data that only holds fast5s